### PR TITLE
hwmv2: Add and use board identifiers for boards with revisions

### PR DESCRIPTION
**Pull request by nordicjm**

### CODE REVIEW

**Review by tejlmand — APPROVED** [review]

**Comment by nordicjm** [comment]
@erwango there is a problem with stm32f411e_disco, it specifies revisions B and D in board.yml but has revision files for B and E

File: boards/st/stm32f411e_disco/stm32f411e_disco_E.yaml
Comment (erwango) [inline boards/st/stm32f411e_disco/stm32f411e_disco_E.yaml:1]: This should be boards/st/stm32f411e_disco/stm32f411e_disco_D.yaml.
Not sure where the 'e' is coming from ?

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/st/stm32f411e_disco/stm32f411e_disco_E.yaml
Comment (nordicjm) [reply boards/st/stm32f411e_disco/stm32f411e_disco_E.yaml:1]: must have been mixed up in documents, fixed

**Comment by erwango** [comment]
Question on the change :
Does this mean that there is no default anymore ?

My worry is this one:
We're using a board_foo since long time which has no revision. It's name (board_foo) is spread in multiple yaml files (tests, samples, boards/foo.overlays).
Someone adds a new revision to this board (board_foo@2). Should we then update all references (s/board_foo/board_foo@1) ? Or can we just keep the initial revision as the default and keep using board_foo ?

**Comment by nordicjm** [comment]
> Question on the change : Does this mean that there is no default anymore ?
> 
> My concern is this one: We're using a board_foo since long time which has no revision. It's name (board_foo) is spread in multiple yaml files (tests, samples, boards/foo.overlays). Someone adds a new revision to this board (board_foo@2). Should we then update all references (s/board_foo/board_foo@1) ? Or can we just keep the initial revision as the default and keep using board_foo ?

No change is needed, so the change here is just in the identifiers which is for twister, the default revision remains the same and building without a revision will work the same as it did before this change, as well as the same overlays being applied. Changes here just affect twister and tests which addresses an issue with clashing twister paths and allows for easier issue tracking if a board fails to know what the actual revision that is failing is

**Review by erwango — APPROVED** [review]

**Pull request merged**